lib/mistral_ai_summary_example/mistralai_summary_page.dart

```diff
@@ -1,13 +1,165 @@
 import 'package:flutter/material.dart';
+import 'package:mistral_ai_chat_example_app/l10n/l10n.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_summary_example/summary_settings.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_summary_example/summary_settings_page.dart';
+import 'package:mistral_ai_chat_example_app/mistral_ai_summary_example/utils.dart';
+import 'package:mistralai_client_dart/mistralai_client_dart.dart';
 
-class MistralAISummaryPage extends StatelessWidget {
+class MistralAISummaryPage extends StatefulWidget {
   const MistralAISummaryPage({super.key});
 
+  @override
+  State<MistralAISummaryPage> createState() => _MistralAISummaryPageState();
+}
+
+class _MistralAISummaryPageState extends State<MistralAISummaryPage> {
+  final MistralAIClient mistralAIClient = MistralAIClient(
+    apiKey: '',
+  );
+
+  final TextEditingController summaryInputController = TextEditingController();
+
+  String summaryResult = '';
+
+  SummarySettings summarySettings = SummarySettings();
+
+  Future<void> summarizeText(String text) async {
+    setState(() {
+      summaryResult = 'Loading...';
+    });
+    try {
+      final response = await mistralAIClient.chat(
+        ChatParams(
+          model: 'mistral-small',
+          temperature: summarySettings.temperature,
+          topP: summarySettings.topP,
+          randomSeed: summarySettings.randomSeed,
+          maxTokens: summarySettings.maxTokens,
+          safePrompt: summarySettings.safePrompt,
+          messages: [
+            ChatMessage(role: 'user', content: text),
+          ],
+        ),
+      );
+      setState(() {
+        summaryResult = response.choices.first.message.content;
+      });
+    } catch (e) {
+      setState(() {
+        summaryResult = 'Error: $e';
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.summaryPageTitle),
+        actions: [
+          IconButton(
+            onPressed: () async {
+              final newSettings = await Navigator.push(
+                context,
+                MaterialPageRoute<SummarySettings>(
+                  builder: (BuildContext context) => SettingsWidget(
+                    mistralAIClient: mistralAIClient,
+                    initialSettings: summarySettings,
+                  ),
+                ),
+              );
+              setState(() {
+                summarySettings = newSettings ?? summarySettings;
+              });
+            },
+            icon: const Icon(Icons.settings),
+          ),
+        ],
+      ),
+      body: SafeArea(
+        child: Padding(
+          padding: const EdgeInsets.all(16),
+          child: SingleChildScrollView(
+            child: Column(
```

Comment:
Could be replaced with `ListView`

Reply:
Resolved in [e14ac83](https://github.com/nomtek/flutter_ai_examples/pull/3/commits/e14ac83905cc3e38b86340e800b12bca80f2ca8d)